src/main/java/com/teragrep/aer_02/DefaultOutput.java

```diff
@@ -0,0 +1,221 @@
+import java.nio.channels.UnresolvedAddressException;
+import java.util.concurrent.TimeoutException;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+// TODO unify, this is a copy from cfe_35 which is a copy from rlo_10 with FIXES
+final class DefaultOutput implements Output {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultOutput.class);
+
+    private final RelpConnection relpConnection;
+    private final String relpAddress;
+    private final int relpPort;
+    private final int reconnectInterval;
+
+    // metrics
+    private final Counter records;
+    private final Counter bytes;
+    private final Counter resends;
+    private final Counter connects;
+    private final Counter retriedConnects;
+    private final Timer sendLatency;
+    private final Timer connectLatency;
+
+    DefaultOutput(String name, RelpConfig relpConfig, MetricRegistry metricRegistry) {
+        this(name, relpConfig, metricRegistry, new RelpConnection());
+    }
+
+    DefaultOutput(String name, RelpConfig relpConfig, MetricRegistry metricRegistry, RelpConnection relpConnection) {
+        this(
+                name,
+                relpConfig,
+                metricRegistry,
+                relpConnection,
+                new SlidingWindowReservoir(10000),
+                new SlidingWindowReservoir(10000)
+        );
+    }
+
+    DefaultOutput(
+            String name,
+            RelpConfig relpConfig,
+            MetricRegistry metricRegistry,
+            RelpConnection relpConnection,
+            Reservoir sendReservoir,
+            Reservoir connectReservoir
+    ) {
+        this.relpAddress = relpConfig.destinationAddress;
+        this.relpPort = relpConfig.destinationPort;
+        this.reconnectInterval = relpConfig.reconnectInterval;
+
+        this.relpConnection = relpConnection;
+        this.relpConnection.setConnectionTimeout(relpConfig.connectionTimeout);
+        this.relpConnection.setReadTimeout(relpConfig.readTimeout);
+        this.relpConnection.setWriteTimeout(relpConfig.writeTimeout);
+
+        this.records = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "records"));
+        this.bytes = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "bytes"));
+        this.resends = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "resends"));
+        this.connects = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "connects"));
+        this.retriedConnects = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "retriedConnects"));
+        this.sendLatency = metricRegistry
+                .timer(name(DefaultOutput.class, "<[" + name + "]>", "sendLatency"), () -> new Timer(sendReservoir));
+        this.connectLatency = metricRegistry
+                .timer(name(DefaultOutput.class, "<[" + name + "]>", "connectLatency"), () -> new Timer(connectReservoir));
+
+        connect();
```

Comment:
Ctor is not code free, so can this be refactored to use **at least** the connect method outside the ctor?